File: golangLeetcode/121.买卖股票的最佳时机.go
/*
 * @lc app=leetcode.cn id=121 lang=golang
 *
 * [121] 买卖股票的最佳时机
 */

// @lc code=start
func maxProfit(prices []int) int {
	if len(prices) <= 1 {
		return 0
	}
	minVal := prices[0]
	sumVal := 0
	for _, v := range prices {
		if minVal < v {
			tmp := v - minVal
			if tmp > sumVal {
				sumVal = tmp
			}
		} else {
			minVal = v
		}
	}
	return sumVal
}

// @lc code=end

File: LoveSwiftLeetcode/121.买卖股票的最佳时机.swift
/*
 * @lc app=leetcode.cn id=121 lang=swift
 *
 * [121] 买卖股票的最佳时机
 */

// @lc code=start
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        var profit = 0
        if prices.count == 0 {
            return 0
        }
        var leftMin = prices[0]
        for i in prices {
            leftMin = min(i, leftMin)
            profit = max(profit, i - leftMin)
        }
        return profit
    }
}
// @lc code=end

